### PR TITLE
RD-3207 Add backend endpoint for generating Terraform blueprint

### PR DESCRIPTION
**Pull request by qooban**
## Description
Adds `POST /terraform/blueprint` endpoint providing Terraform blueprint generation. No validation and error handling implemented as this is planned for the other tickets.

## Screenshots / Videos
N/A

## Checklist
- [x] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires a documentation update.
- [x] I added proper labels to this PR.

## Tests
Added backend tests for `POST /terraform/blueprint` enpdoint. 

## Documentation
Updated backend architecture diagrams.

### CODE REVIEW

File: backend/app.ts [hunk 1]
```diff
@@ -101,27 +102,28 @@ app.use(
 );
 
 // API Routes
-app.use(`${contextPath}/sp`, ServerProxy);
```
Comment (qooban) [inline backend/app.ts:104]: This is just order line order change (sorted them alphabetically) + adding `app.use(`${contextPath}/tf`, Terraform);` line.

File: backend/routes/Terraform.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            variables,
+            environmentVariables,
+            outputs
+        },
+        {
+            views: [templatePath]
+        }
+    );
+
+    res.setHeader('content-type', 'text/x-yaml');
```
Comment (qooban) [inline backend/routes/Terraform.ts:71]: I'm not sure about `text/x-yaml`. This is just example of MIME type I found in the net.
Any other suggestions are welcome.

File: backend/templates/terraform/blueprint.ejs [hunk 3]
```diff
@@ -0,0 +1,49 @@
+tosca_definitions_version: cloudify_dsl_1_3
+
+description: >
+  This blueprint creates infrastructure on AWS using Terraform.
```
Comment (qooban) [inline backend/templates/terraform/blueprint.ejs:4]: This probably will change. I didn't get final info about the description in the generated blueprint.

File: backend/routes/Terraform.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+interface RequestBody {
+    terraformVersion: string;
+    terraformTemplate: string;
+    resourceLocation: string;
+    variables?: Variable[];
+    environmentVariables?: Variable[];
+    outputs?: Output[];
+}
```
Comment (qooban) [inline backend/routes/Terraform.ts:37]: Depending on how will it be used, we can move it to different place. Maybe that's good time to start reusing data between stage frontend and stage backend?

Comment (kubama) [reply backend/routes/Terraform.ts:37]: Would be great to start sharing code between the two, I guess it's up to you and your capacity, alternatively I could try to handle that as part of frontend work.

**Review by NorbertAmartus — APPROVED** [review]
Looks good 💪 
I have nothing more to add to the scope of the started discussions

File: backend/app.ts [hunk 5]
```diff
@@ -101,27 +102,28 @@ app.use(
...
+app.use(`${contextPath}/file`, File);
+app.use(`${contextPath}/filters`, Filters);
+app.use(`${contextPath}/github`, GitHub);
 app.use(`${contextPath}/maps`, Maps);
+app.use(`${contextPath}/plugins`, Plugins);
+app.use(`${contextPath}/source`, SourceBrowser);
+app.use(`${contextPath}/sp`, ServerProxy);
+app.use(`${contextPath}/style`, Style);
+app.use(`${contextPath}/templates`, Templates);
+app.use(`${contextPath}/tf`, Terraform);
```
[the remark this answers is not in the document]
Comment (qooban) [reply backend/app.ts]: Make sense, changed.

File: doc/diagrams/architecture-backend.txt [hunk 6]
```diff
@@ -1,8 +1,9 @@
 [note:Cloudify Stage Backend Architecture {bg:indianred}]
 
-[note: Frontend {bg:forestgreen}]<--HTTP-->[ExpressJS Web Server|server.js {bg:red}]
```
Comment (qooban) [inline doc/diagrams/architecture-backend.txt:3]: Adding labels to edges stopped working in https://yuml.me.

**Review by qooban — COMMENTED** [review]
Updated backend architecture diagram and removed diagrams `*.png` files as I think having text version is enough.

File: backend/routes/Terraform.ts (hunk 4 above)
Comment (qooban) [reply backend/routes/Terraform.ts:37]: I will leave it for now. Don't have capacity to provide future-proof solution.

**Review by kubama — APPROVED** [review]

**Pull request merged**